### ISSUE TITLE
Auter errors when saved patches are already installed

### ISSUE THREAD

**Issue by reaperzn** [rackerlabs/auter#31]
If we have the ONLYINSTALLFROMPREP set to yes we can get into the following situation:

1) auter --prep is run which downloads the rpm files locally
2) auter --apply is run or a user manually updates the system which includes all the downloaded packages
3) auter --apply is run a second time or after the manual update, you will get the following error:
```
[root@paolo-auter-test7 ~]# auter --apply
INFO: Running with: /usr/bin/auter --apply
INFO: Running in an interactive shell, disabling all random sleeps
INFO: Applying updates
ERROR: Updates failed. Exiting.
```
There are a few issues here:

1) We are relying on a directory listing to deturmin if there are updates to be applied:
```
  if [[ "${ONLYINSTALLFROMPREP}" == "yes" ]]; then
    RC=0
    if [[ $(ls -1 ${DOWNLOADDIR}/${CONFIGSET}/*.rpm | wc -l) -gt 0 ]]; then
      RPMS="${DOWNLOADDIR}/${CONFIGSET}/*.rpm"
      RC=100
    fi
    # When passing RPMs to dnf/yum, the update verb won't install any that aren't already
    # installed (i.e. dependencies of other packages). Instead we need to use install.
    UPDATEACTION="install"
  else
    local RC=$(${PACKAGE_MANAGER} check-update ${PACKAGEMANAGEROPTIONS} &>/dev/null; echo $?)
    UPDATEACTION="update"
  fi
```

2) The actual error is because of the following lines of code in the apply_updates function:
```
    local HISTORY_BEFORE=$(${PACKAGE_MANAGER} history list)
...
...
    local HISTORY_AFTER=$(${PACKAGE_MANAGER} history list)

    if [[ "${HISTORY_BEFORE}" == "${HISTORY_AFTER}" ]]; then
      logit "ERROR: Updates failed. Exiting."
      quit 3
    fi
```
Example:
```[root@paolo-auter-test7 ~]# auter --prep
INFO: Running with: /usr/bin/auter --prep
INFO: Running in an interactive shell, disabling all random sleeps
INFO: Updates downloaded to /var/cache/auter/default


[root@paolo-auter-test7 ~]# ll /var/cache/auter/default
total 1536
-rw-r--r-- 1 root root 448108 Jan 18 12:57 openssh-6.6.1p1-33.el7_3.x86_64.rpm
-rw-r--r-- 1 root root 657552 Jan 18 12:58 openssh-clients-6.6.1p1-33.el7_3.x86_64.rpm
-rw-r--r-- 1 root root 450388 Jan 18 12:58 openssh-server-6.6.1p1-33.el7_3.x86_64.rpm

[root@paolo-auter-test7 ~]# auter --apply
INFO: Running with: /usr/bin/auter --apply
INFO: Running in an interactive shell, disabling all random sleeps
INFO: Applying updates
INFO: Updates complete (yum Transaction ID : 20). You may need to reboot for some updates to take effect


[root@paolo-auter-test7 ~]# auter --apply
INFO: Running with: /usr/bin/auter --apply
INFO: Running in an interactive shell, disabling all random sleeps
INFO: Applying updates
ERROR: Updates failed. Exiting.
```

To remedy this in the following way:
```
if [[ "${ONLYINSTALLFROMPREP}" == "yes" ]]; then
  RC=0
    if [[ $(ls -1 ${DOWNLOADDIR}/${CONFIGSET}/*.rpm | wc -l) -gt 0 ]]; then
      RPMS="${DOWNLOADDIR}/${CONFIGSET}/*.rpm"
      RC=$(${PACKAGE_MANAGER} check-update $(find ${DOWNLOADDIR}/${CONFIGSET}/*.rpm -printf "%f\n" | sed 's/.rpm$//g' ))
    fi
    # When passing RPMs to dnf/yum, the update verb won't install any that aren't already
    # installed (i.e. dependencies of other packages). Instead we need to use install.
    UPDATEACTION="install"
  else
    local RC=$(${PACKAGE_MANAGER} check-update ${PACKAGEMANAGEROPTIONS} &>/dev/null; echo $?)
    UPDATEACTION="update"
fi
```

Any thoughts or alternative suggestions?

**Comment by reaperzn** [comment]
Can someone review this?

**Comment by reaperzn** [comment]
Fixed in #48
